src/Spout/Writer/XLSX/Helper/StyleHelper.php

```diff
@@ -71,6 +82,32 @@ protected function registerFill($style)
         }
     }
 
+    /**
+     * Register a border definition
+     *
+     * @param $style
+     */
+    protected function registerBorder($style)
+    {
+        $styleId = $style->getId();
+
+        if (true === $style->shouldApplyBorder()) {
+
+            $border = $style->getBorder();
+            $serializedBorder = serialize($border);
+
+            if (!isset($this->registeredBorders[$serializedBorder])) {
+
+                $this->registeredBorders[$serializedBorder] = $styleId;
+                $this->styleIdToBorderMappingTable[$styleId] = count($this->registeredBorders);
+
+            }
+
+        } else {
+            $this->styleIdToBorderMappingTable[$styleId] = 0;
```

Comment:
Hi, thanks for the review.

The 0 depicts - that if there is no border explicitly set - the borderId is 0.

``` <xf numFmtId="0" fontId="3" fillId="0" borderId="0" xfId="0" applyFont="1" applyBorder="0"/>```

The first border is "no border" in the list (0 based): ```<border><left/><right/><top/><bottom/></border>```

This does not necessarily has to happen at this point - but it should somewhere.

Regarding the sharing of definitions. mea culpa - I commited a failing test. I'll look into it shortly.